app/server/lib/requestUtils.ts

```diff
@@ -100,15 +101,25 @@ export function allowHost(req: Request, allowedHost: string|URL) {
   const proto = getEndUserProtocol(req);
   const actualUrl = new URL(getOriginUrl(req));
   const allowedUrl = (typeof allowedHost === 'string') ? new URL(`${proto}://${allowedHost}`) : allowedHost;
-  if (mreq.isCustomHost) {
-    // For a request to a custom domain, the full hostname must match.
-    return actualUrl.hostname === allowedUrl.hostname;
-  } else {
-    // For requests to a native subdomains, only the base domain needs to match.
-    const allowedDomain = parseSubdomain(allowedUrl.hostname);
-    const actualDomain = parseSubdomain(actualUrl.hostname);
-    return (actualDomain.base === allowedDomain.base);
-  }
+
+  if (mreq.isCustomHost && actualUrl.hostname === allowedUrl.hostname) { return true; }
+
+  // For requests to a native subdomains, only the base domain needs to match.
+  const allowedDomain = parseSubdomain(allowedUrl.hostname);
+  const actualDomain = parseSubdomain(actualUrl.hostname);
+  if (!_.isEmpty(actualDomain) && actualDomain.base === allowedDomain.base) { return true; }
```

Comment:
The comment for this function says "differing ports are allowed, which helps in dev/testing".  Is that still true? Should the comment be updated?

The addition of `GRIST_ALLOWED_HOSTS: 'localhost:8095'` bothers me a bit, I'd have expected that setups without `GRIST_ALLOWED_HOSTS` would continue to work as before, and that this PR would be offering a way to strictly expand trust rather than reduce it.

Reply:
You're perfectly right. This morning, it bothered me and I hoped you did merged it. I'll try another way and not touch the allow host function.

Reply:
> The comment for this function says "differing ports are allowed, which helps in dev/testing". Is that still true? Should the comment be updated?

It seems to be still true.